parsers/src/main/java/org/coode/xml/XMLWriterImpl.java

```diff
@@ -64,6 +64,7 @@
  */
 public class XMLWriterImpl implements XMLWriter {
 
+    private final XMLWriterPreferences instance = XMLWriterPreferences.getInstance();
```

Comment:
I don't think so. Given how things blew up when one thread changed the use entities setting whilst another class was writing its output (thus causing it to assume that it had defined entities when starting the document, and thus start using entities that were in fact undefined). 

  The reason for extracting  to a field is that the preference object is accessed for every line written to check for indentation settings.  
The XML formats have a lot of lines.